### PR TITLE
[DA-216] Add endpoint to fill the whitelist from the product's BOM

### PR DESCRIPTION
**Pull request by jomrazek**

### CODE REVIEW

File: communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java [hunk 1]
```diff
@@ -125,6 +125,10 @@ public GAVDependencyTree getDependencyTreeOfGAV(GAV gav) throws CommunicationExc
 
     @Override
     public Optional<MavenProject> getPom(GAV gav) throws CommunicationException {
+        return pomAnalyzer.readPom(getPomStream(gav).get());
```
Comment (jbrazdil) [inline communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java]: there should be something like:
```java
return getPomStream(gav).map(x -> pomAnalyzer.readPom(x));
```

**Comment by jbartece** [comment]
:+1:

File: communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzerImpl.java [hunk 2]
```diff
@@ -186,6 +201,59 @@ private GAV generateGAV(ProjectVersionRef ref) {
         return new GAV(ref.getGroupId(), ref.getArtifactId(), ref.getVersionString());
     }
 
+    public List<MavenPomView> getGitPomView(String scmUrl, String revision, String pomPath,
```
Comment (jbrazdil) [inline communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzerImpl.java]: @Override

File: communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzerImpl.java [hunk 3]
```diff
@@ -186,6 +201,59 @@ private GAV generateGAV(ProjectVersionRef ref) {
...
+            for (Map.Entry<File, ProjectVersionRef> entry : projectVersionRefs.entrySet()) {
+                poms.add(mavenPomReader.read(entry.getValue(),
+                        getRepoLocations(tempDir, clonedDir, repositories)));
+            }
+
+            return poms;
+        } catch (ScmException | IOException | GalleyMavenException e) {
+            log.warn(e.getMessage());
+        } finally {
+            FileUtils.deleteQuietly(clonedDir);
```
Comment (jbrazdil) [inline communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzerImpl.java]: SCM class take care of clonedDir. If you delete it manually, you can compromise cache.

File: reports-backend/src/main/java/org/jboss/da/listings/impl/service/WLFillerImpl.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    @Override
+    public WLStatus fillWhitelistFromPom(String scmUrl, String revision, String pomPath,
+            List<String> repositories, long productId) {
+        if (!checkProduct(productId)) {
+            return WLStatus.PRODUCT_NOT_FOUND;
+        }
+        try {
+            fillWLFromList(getPomListFromGit(scmUrl, revision, pomPath, repositories), productId);
+        } catch (GalleyMavenException | PomAnalysisException e) {
+            return WLStatus.ANALYSER_ERROR;
```
Comment (jbrazdil) [inline reports-backend/src/main/java/org/jboss/da/listings/impl/service/WLFillerImpl.java]: Maybe log the error?

File: reports-backend/src/main/java/org/jboss/da/listings/impl/service/WLFillerImpl.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        try {
+            Optional<InputStream> is = aprox.getPomStream(new GAV(groupId, artifactId, version));
+            if (is.isPresent()) {
+                MavenPomView view = analyzer.getMavenPomView(is.get());
+                fillWLFromList(Collections.singletonList(view), productId);
+            } else {
+                return WLStatus.ANALYSER_ERROR;
+            }
+        } catch (CommunicationException | ConfigurationParseException | GalleyMavenException e) {
+            return WLStatus.ANALYSER_ERROR;
```
Comment (jbrazdil) [inline reports-backend/src/main/java/org/jboss/da/listings/impl/service/WLFillerImpl.java]: Maybe log the error?

**Comment by jbrazdil** [comment]
Why the rest endpoint takes pomPath and doesn't use it anywhere?

**Comment by jomrazek** [comment]
retest this please

**Comment by jomrazek** [comment]
test this please

**Comment by jomrazek** [comment]
retest this please

**Comment by jomrazek** [comment]
retest this please

**Comment by jomrazek** [comment]
retest this please

**Comment by jomrazek** [comment]
retest this please

**Comment by jbartece** [comment]
:+1:

**Pull request merged**